### ISSUE TITLE
Upgrade to v8

### ISSUE THREAD

**Issue by ismailmayat** [scyllagroup/UmbracoFlare#16]
Hello,

Are there any plans to update this package to v8?  If not we at Novicell may look to fork and update.

Regards

Ismail

**Comment by markadrake** [comment]
@ismailmayat I'm going to bring this up to our team tomorrow and report back. I'll give you an update with our expectations moving forward. 

The reason we haven't already upgraded this to support Umbraco 8 is that we have both support and future projects scheduled and those projects are already locked into Umbraco 7. 

Have you and your team started any such work? 

All the best!

**Comment by ismailmayat** [comment]
@markadrake we have not started any work on it, i was just going to rip parts of the on publish code to clear out cache.

**Comment by markadrake** [comment]
@ismailmayat I don't have an ETA for the update, but we started work on converting this for Umbraco v8 and I will continue to monitor and update you here on the progress.

**Comment by ismailmayat** [comment]
Mark,

Awesome many thanks, we are looking to go live around with new site that will use cloudflare jan 2020

**Comment by ismailmayat** [comment]
Not sure if this helps but I have recently been using https://github.com/dawoe/umbraco-nexu which is now out for v8, it allows you to pick up item usages, with UmbracoFlare I guess that is the main issue figuring out where stuff is used so that you can invalidate things correctly.

Maybe worth looking at this for v8 umbracoflare as all the look up of usages is already done for you?

**Comment by brreisner** [comment]
Any updates on progress here? Very much looking forward to testing this in v8

**Comment by markadrake** [comment]
@jmcaveney and @agriffi10 wondering if you guys have made the UI update needed for the next release of UmbracoFlare? See comments above. All the best!

**Comment by markadrake** [comment]
@brreisner last I heard we had v8 support but the user interface needed some work. As you might have already seen, [there is a branch with some work done already](https://github.com/scyllagroup/UmbracoFlare/tree/dev-v8), some 33 commits ahead of master at the time of this response. I'll see if I can work with them to bring this up to date.

**Comment by ismailmayat** [comment]
Guys,

I took a copy of the branch and removed all the ui stuff which i did not need and made few tweaks here and there and I have it working. The project is going live in next 2 months.

**Comment by jmcaveney** [comment]
> 
> 
> Guys,
> 
> I took a copy of the branch and removed all the ui stuff which i did not need and made few tweaks here and there and I have it working. The project is going live in next 2 months.

@ismailmayat , can you tell me what "tweaks" you made or maybe send us a PR? I'm happy destroying the UI as it exists now and rebuilding it, it was completely overdone.

**Comment by jmcaveney** [comment]
> 
> 
> @jmcaveney and @agriffi10 wondering if you guys have made the UI update needed for the next release of UmbracoFlare? See comments above. All the best!

@markadrake not yet. send us a PR if you're interested.

**Comment by ismailmayat** [comment]
@jmcaveney intially i forked but after downloading i just put the umbracoflare as a separate project into my project.  I have however uploaded new repo https://github.com/ismailmayat/UmbracoFlare with all my changes.  I only use events and the purge cache tree option.

I added extra config property ShowPurgeMenu to show hide that option but keep the PurgeCacheOn option separate so that events would still work.  In my one I also stripped out all the multi lingual url stuff as i didnt need it for this site (was rushing to get through stuff to be honest client had tight budget, also brexit the world speaks english right ;-} )

Any questions fire away more than happy to help as this all works very nicely for me.

**Comment by jmcaveney** [comment]
> @jmcaveney intially i forked but after downloading i just put the umbracoflare as a separate project into my project. I have however uploaded new repo https://github.com/ismailmayat/UmbracoFlare with all my changes. I only use events and the purge cache tree option.
> 
> I added extra config property ShowPurgeMenu to show hide that option but keep the PurgeCacheOn option separate so that events would still work. In my one I also stripped out all the multi lingual url stuff as i didnt need it for this site (was rushing to get through stuff to be honest client had tight budget, also brexit the world speaks english right ;-} )
> 
> Any questions fire away more than happy to help as this all works very nicely for me.

Great thanks! I will try to review this today. Will ping you on twitter with any ?s.

**Comment by ismailmayat** [comment]
@jmcaveney cool no worries

**Comment by brreisner** [comment]
Any update here?  Very interested in using this for v8

**Comment by brreisner** [comment]
@jmcaveney any updates here?

**Comment by agriffi10** [comment]
> @jmcaveney any updates here?

Hi there @brreisner , I'm also on the Scylla team, and all I can tell you is it's in our backlog. We do not yet have a hard timeline for this to be migrated.

**Comment by brreisner** [comment]
Appreciate the response. Look forward to a v8 release when available. I'm assuming your team will update this thread with any updates?

**Comment by agriffi10** [comment]
Absolutely! We are currently in the process of a major refactor of our various internal Umbraco tools (non breaking, just a lot of clean up that had been put off due to time constraints), and we're moving through that at a nice pace. Once that is finished we're going to start upgrading everything to v8. Our tentative timeline is to start the v8 process by this summer.

**Comment by max-mayorov** [comment]
Hi everyone! Is there any progress on this?

**Comment by agriffi10** [comment]
@mcsdwarken Yes, there is. While progress is slow, we're in the process of upgrading things to v8.

**Comment by markadrake** [comment]
At least with this latest release of Umbraco (8.7) there is a clear upgrade path for your clients who were on v7 and dependent on InnerContent and StackedContent packages for their editing experience. Last I checked with this project, it "worked" with v8 but needed some UI love?

**Comment by barrar** [comment]
I've made some progress updating UmbracoFlare for 8.7.0, any chance your have more recent commits for the `dev-v8` branch?
